misc/scripts/black_format.sh

```diff
@@ -20,7 +20,7 @@ fi
 # A diff has been created, notify the user, clean up, and exit.
 printf "\n\e[1;33m*** The following changes must be made to comply with the formatting rules:\e[0m\n\n"
 # Perl commands replace trailing spaces with `·` and tabs with `<TAB>`.
-printf "$diff\n" | perl -pe 's/(.*[^ ])( +)(\e\[m)$/my $spaces="·" x length($2); sprintf("$1$spaces$3")/ge' | perl -pe 's/(.*[^\t])(\t+)(\e\[m)$/my $tabs="<TAB>" x length($2); sprintf("$1$tabs$3")/ge'
+echo "$diff" | perl -pe 's/(.*[^ ])( +)(\e\[m)$/my $spaces="·" x length($2); sprintf("$1$spaces$3")/ge' | perl -pe 's/(.*[^\t])(\t+)(\e\[m)$/my $tabs="<TAB>" x length($2); sprintf("$1$tabs$3")/ge'
```

Comment:
Could you try `printf "%s\n" "$diff"`  instead of echo?

Reply:
Can try!

Reply:
Thank you that does work! Will apply